### PR TITLE
Fixes #788 - Remove ClientInfo from API

### LINKED ISSUE #788: Remove ClientInfo from API
`ClientInfo` isn't really necessary in the API as `Properties` can be used instead in most cases. I would like to remove it from the API and make the following changes:

* Remove `changeUser()` and `info()` from AccumuloClient
* Use `Properties` where `ClientInfo` is used
* Keep `ClientInfo` as internal class
* Add `properties()` and `token()` to AccumuoClient. `token()` is needed for code that creates `ClientContext` from `AccumuloClient`.
* Stop creating `ClientInfo` from `Accumulo.newClient()` builder and create a `Accumulo.newProperties()` builder.

| Current | New |
|---------|-----|
| ClientInfo info = client.info() | Properties props = client.properties(); Authentication token = client.token() |
| client.changerUser(user, token) | Accumulo.newClient().from(oldClient.properties()).as(user, token).build() |
| ClientInfo info = Accumulo.newClient().to(instance, zoo).from(user, token).info() | Properties props = Accumulo.newProperties().to(instance, zoo).from(user, token).build() |

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java [hunk 1]
```diff
@@ -31,7 +31,6 @@
 import org.apache.accumulo.core.client.AccumuloException;
 import org.apache.accumulo.core.client.AccumuloSecurityException;
 import org.apache.accumulo.core.client.BatchWriterConfig;
-import org.apache.accumulo.core.client.ClientInfo;
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java]: Unused import?

Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java]: Oh right its now in impl, so you dont need the import.

**Comment by mikewalch** [comment]
Tests are failing for this PR due to Apilyzer:

```
Problems : 

  CONTEXT              TYPE                                                         FIELD/METHOD                        NON-PUBLIC REFERENCE

  METHOD_PARAM         org.apache.accumulo.core.client.mapreduce.AbstractInputFormat setClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_RETURN        org.apache.accumulo.core.client.mapreduce.AbstractInputFormat getClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_PARAM         org.apache.accumulo.core.client.mapreduce.AccumuloOutputFormat setClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_RETURN        org.apache.accumulo.core.client.mapreduce.AccumuloOutputFormat getClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_PARAM         org.apache.accumulo.core.client.mapred.AbstractInputFormat   setClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_RETURN        org.apache.accumulo.core.client.mapred.AbstractInputFormat   getClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_PARAM         org.apache.accumulo.core.client.mapred.AccumuloOutputFormat  setClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
  METHOD_RETURN        org.apache.accumulo.core.client.mapred.AccumuloOutputFormat  getClientInfo(...)                  org.apache.accumulo.core.clientImpl.ClientInfo
```

ClientInfo need to be removed from M/R API before this can merged.  So this PR is dependent on #804

File: core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java (hunk 1 above)
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java]: impsort-maven-plugin FTW! :smiley_cat:

**Comment by mikewalch** [comment]
I updated this pull request to remove ClientInfo from M/R code

**Pull request merged**